superset-frontend/src/components/Chart/ChartRenderer.jsx

```diff
@@ -92,13 +113,16 @@ class ChartRenderer extends React.Component {
     };
   }
 
-  shouldComponentUpdate(nextProps) {
+  shouldComponentUpdate(nextProps, nextState) {
     const resultsReady =
       nextProps.queriesResponse &&
       ['success', 'rendered'].indexOf(nextProps.chartStatus) > -1 &&
       !nextProps.queriesResponse?.[0]?.error;
 
     if (resultsReady) {
+      if (!isEqual(this.state, nextState)) {
+        return true;
+      }
```

Comment:
```suggestion
     return !isEqual(this.state, nextState)
```

Reply:
I don't want to return unless the state changes. If the state is the same, it should execute the code below these lines. 

```
if (!isEqual(this.state, nextState)) {
   return true;
}

this.hasQueryResponseChange = nextProps.queriesResponse !== this.props.queriesResponse;
...
```